src/os_db.ml

```diff
@@ -60,12 +59,23 @@ let connect () = Lwt_PGOCaml.connect
   ?unix_domain_socket_dir:!unix_domain_socket_dir_r
   ()
 
-let validate db =
-  try_lwt
-    lwt () = Lwt_PGOCaml.ping db in
-    Lwt.return true
-  with _ ->
-    Lwt.return false
+let pool_r = ref @@ Lwt_pool.create 16 ~validate connect
+
+let pool : (string, bool) Hashtbl.t Lwt_PGOCaml.t Lwt_pool.t = !pool_r
```

Comment:
This does not work. We want to always use the most recent pool, but [pool] refers to the initial one.